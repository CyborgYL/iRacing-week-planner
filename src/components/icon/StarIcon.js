// @flow

import React from 'react';
import styles from '../../styles/main.module.scss';

type Props = {

};

export default function StarIcon() {
  return <span className={`${styles.glyphicon} ${styles['glyphicon-star']}`} />;
}
