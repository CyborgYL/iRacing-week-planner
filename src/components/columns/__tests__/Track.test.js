import React from 'react';
import { shallow } from 'enzyme';
import { describe, test } from '@jest/globals';
import StarIcon from '../../icon/StarIcon';

import { Track } from '../index';

describe('components/columns/Track', () => {
  test('renders no own or fave', () => {
    const component = shallow(
      <Track
        race={{
          trackId: 125,
        }}
        ownedTracks={[]}
        favouriteTracks={[]}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.find('.success').length).toBe(0);
    expect(component.find(StarIcon).length).toBe(0);
  });

  test('renders ownership', () => {
    const component = shallow(
      <Track
        race={{
          trackId: 125,
        }}
        ownedTracks={[125]}
        favouriteTracks={[]}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.find('.success').length).toBe(1);
    expect(component.find(StarIcon).length).toBe(0);
  });

  test('renders favourite', () => {
    const component = shallow(
      <Track
        race={{
          trackId: 125,
        }}
        ownedTracks={[]}
        favouriteTracks={[125]}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.find('.success').length).toBe(0);
    expect(component.find(StarIcon).length).toBe(1);
  });
});
