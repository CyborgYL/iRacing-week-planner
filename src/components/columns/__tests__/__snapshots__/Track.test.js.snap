// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/columns/Track renders favourite 1`] = `
<td
  className=""
>
  <div>
    <StarIcon />
    <span>
       
    </span>
  </div>
</td>
`;

exports[`components/columns/Track renders no own or fave 1`] = `
<td
  className=""
>
  <div>
    <span>
       
    </span>
  </div>
</td>
`;

exports[`components/columns/Track renders ownership 1`] = `
<td
  className="success"
>
  <div>
    <span>
       
    </span>
  </div>
</td>
`;
